### PR TITLE
Closes #2050: Automatic Cache Clearing - Do not attempt to clear cache files when do_rocket_generate_caching_files is false

### LINKED ISSUE #2050: Automatic Cache Clearing - Do not attempt to clear cache files when do_rocket_generate_caching_files is false
We set `do_rocket_generate_caching_files` to false for compatibility with various hosts. 

Since no cache files are generated anymore, at this point there is no point in trying to clear the cache files every hour ( #486 )

**Thoughts:** 

- The hourly cron is still potentially useful to trigger the preload. 
- Also firing the `rocket_before_automatic_cache_purge_dir` and `rocket_after_automatic_cache_purge` could be useful for doing other actions. 
- The part we know for sure that is useless is the attempt to delete the files that are not generated in the first place. 

Related discussion: https://wp-media.slack.com/archives/G8L4EE8N9/p1571358335127500

### PR DESCRIPTION
**Pull request by Screenfeed**
When the filter `do_rocket_generate_caching_files` is set to `false`, do not delete cache files (but still trigger surrounding hooks). #2050

### CODE REVIEW

File: inc/classes/Cache/class-expired-cache-purge.php [hunk 1]
```diff
@@ -100,43 +103,45 @@ public function purge_expired_files( $lifespan ) {
...
-			$files       = $this->get_cache_files_in_dir( $file );
 			$url_deleted = [];
 
-			foreach ( $files as $item ) {
-				$dir_path     = $item->getPathname();
-				$sub_dir_path = $dir_path . $sub_dir;
-
-				// Time to cut old leaves.
-				$item_paths = $this->purge_dir( $sub_dir_path, $file_age_limit );
+			if ( $cache_enabled ) {
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/Cache/class-expired-cache-purge.php]: What is `purge_expired_cache`?

[the remark this answers is not in the document]
Comment (crystinutzaa) [reply inc/classes/Cache/class-expired-cache-purge.php]: Sorry... I was reffering to the whole function `purge_expired_files()` should run if cache is not enabled?

[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/Cache/class-expired-cache-purge.php]: According to #2050, the hooks in the function should still be triggered.

[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/Cache/class-expired-cache-purge.php]: Unless I misunderstood.

[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/Cache/class-expired-cache-purge.php]: @crystinutzaa?

[the remark this answers is not in the document]
Comment (crystinutzaa) [reply inc/classes/Cache/class-expired-cache-purge.php]: If the hooks need to be triggered them is ok.

**Pull request merged**